src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/WithdrawableService.kt

```diff
@@ -16,24 +16,25 @@ class WithdrawableService(
   @Lazy private val placementRequestService: PlacementRequestService,
   @Lazy private val bookingService: BookingService,
   @Lazy private val placementApplicationService: PlacementApplicationService,
+  @Lazy private val userAccessService: UserAccessService,
 ) {
   val approvedPremisesWithdrawnByPPBookingWithdrawnReasonId: UUID =
     UUID.fromString("d39572ea-9e42-460c-ae88-b6b30fca0b09")
 
-  fun allWithdrawables(application: ApprovedPremisesApplicationEntity): Withdrawables {
+  fun allWithdrawables(application: ApprovedPremisesApplicationEntity, showAllBookings: Boolean = false): Withdrawables {
     val placementRequests = placementRequestService.getWithdrawablePlacementRequests(application)
     val bookings = bookingService.getCancelleableBookings(application)
     val placementApplications = placementApplicationService.getWithdrawablePlacementApplications(application)
 
     return Withdrawables(
       placementRequests = placementRequests,
-      bookings = bookings,
+      bookings = if (showAllBookings) { bookings } else { bookings.filter { userAccessService.currentUserCanManagePremisesBookings(it.premises) } },
       placementApplications = placementApplications,
     )
   }
 
   fun withdrawAllForApplication(application: ApprovedPremisesApplicationEntity, user: UserEntity) {
-    val withdrawables = allWithdrawables(application)
+    val withdrawables = allWithdrawables(application, true)
```

Comment:
what's the implication of this if the user doesn't have permissions to withdraw a booking that exists? Or will that never happen?

Reply:
This won't happen because the permissions are handled in the controller (we should be much more consistent about where this kind of stuff happens!)